src/app/devices/devices.component.spec.ts

```diff
@@ -18,9 +18,10 @@ describe('DevicesComponent', () => {
   let getDevicesSpy: jasmine.Spy
   let getTagsSpy: jasmine.Spy
   let getPowerStateSpy: jasmine.Spy
+  let sendPowerActionSpy: jasmine.Spy
 
   beforeEach(async () => {
```

Comment:
tagChange, masterToggle, resetResponse, bulkPowerAction tests aren't tested either at all or not completely

![image](https://user-images.githubusercontent.com/22204562/143308016-1fe2ac61-fca7-4aed-9bcd-14c41aecf6b4.png)

Reply:
resolved